internal/system/snapd_client_test.go

```diff
@@ -0,0 +1,362 @@
+		Status:     "OK",
+	}
+	snap := snapdSnap{
+		ID:          "test-id",
+		Name:        "juju",
+		Status:      "active",
+		Version:     "3.6.11",
+		Confinement: "strict",
+	}
+	result, err := json.Marshal(snap)
+	if err != nil {
+		t.Fatalf("Failed to marshal snap: %v", err)
+	}
+	response.Result = result
+
+	var parsedSnap snapdSnap
+	if err := json.Unmarshal(response.Result, &parsedSnap); err != nil {
+		t.Fatalf("Failed to parse snap: %v", err)
+	}
+
+	if parsedSnap.ID != "test-id" {
+		t.Errorf("Expected snap ID 'test-id', got: %s", parsedSnap.ID)
+	}
+	if parsedSnap.Name != "juju" {
+		t.Errorf("Expected snap name 'juju', got: %s", parsedSnap.Name)
+	}
+	if parsedSnap.Status != "active" {
+		t.Errorf("Expected snap status 'active', got: %s", parsedSnap.Status)
+	}
+	if parsedSnap.Version != "3.6.11" {
+		t.Errorf("Expected snap version '3.6.11', got: %s", parsedSnap.Version)
+	}
+	if parsedSnap.Confinement != "strict" {
+		t.Errorf("Expected snap confinement 'strict', got: %s", parsedSnap.Confinement)
+	}
+}
+
+func TestSnapdFindOneParsing(t *testing.T) {
+	response := SnapdResponse{
+		Type:       "sync",
+		StatusCode: 200,
+		Status:     "OK",
+	}
+	snaps := []snapdSnap{
+		{
+			ID:          "test-id",
+			Name:        "juju",
+			Version:     "3.6.11",
+			Confinement: "strict",
+			Channels: map[string]ChannelInfo{
+				"3.6/stable": {
+					Confinement: "strict",
+					Version:     "3.6.11",
+				},
+				"2.9/stable": {
+					Confinement: "classic",
+					Version:     "2.9.52",
+				},
+			},
+		},
+	}
+	result, err := json.Marshal(snaps)
+	if err != nil {
+		t.Fatalf("Failed to marshal snaps: %v", err)
+	}
+	response.Result = result
+
+	var parsedSnaps []snapdSnap
+	if err := json.Unmarshal(response.Result, &parsedSnaps); err != nil {
+		t.Fatalf("Failed to parse snaps: %v", err)
+	}
+
+	if len(parsedSnaps) != 1 {
+		t.Fatalf("Expected 1 snap, got: %d", len(parsedSnaps))
+	}
+
+	snap := parsedSnaps[0]
+
+	if snap.Name != "juju" {
+		t.Errorf("Expected snap name 'juju', got: %s", snap.Name)
+	}
+	if len(snap.Channels) != 2 {
+		t.Errorf("Expected 2 channels, got: %d", len(snap.Channels))
+	}
+	if snap.Channels["2.9/stable"].Confinement != "classic" {
+		t.Errorf("Expected 2.9/stable to be classic, got: %s", snap.Channels["2.9/stable"].Confinement)
+	}
+	if snap.Channels["3.6/stable"].Confinement != "strict" {
+		t.Errorf("Expected 3.6/stable to be strict, got: %s", snap.Channels["3.6/stable"].Confinement)
+	}
+}
+
+// TestNewSnapdClient ensures the client is created correctly.
+func TestNewSnapdClient(t *testing.T) {
+	client := NewSnapdClient()
+	if client == nil {
+		t.Fatal("Expected non-nil client")
+	}
+	if client.httpClient == nil {
+		t.Fatal("Expected non-nil HTTP client")
+	}
+}
+
+// TestSnapdClient_Integration_RealSocket is an integration test that only runs if snapd socket is available.
+func TestSnapdClient_Integration_RealSocket(t *testing.T) {
+	if _, err := os.Stat("/run/snapd.socket"); os.IsNotExist(err) {
```

Comment:
I think we shouldn't add integration-level tests here against real snapd. That seems like asking for trouble.

I'd prefer this done at more the "unit" level, by allowing us to configure the socket path. The best way would be to add a `Config` struct that you pass to `NewClient`, with a `Socket string` field like the official client has, and if that's `!=""` use it, otherwise use the default. Then in tests we create a simple socket server using `httptest` with a custom Listener on a socket in a temporary directory, and point Config.Socket to that.

This is how we test Pebble and snapd.